### PR TITLE
Provide universal extension API for host functions.

### PR DESCRIPTION
**Pull request by olonho**

### CODE REVIEW

File: runtime/near-vm-runner-standalone/example-contract/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,8 @@ serde = { version = "1", features = ["derive"] }
 serde_json = "1"
 wee_alloc = "0.4.5"
 
-near-sdk = "0.6.0"
+#near-sdk = "0.6.0"
+near-sdk = { path = "/Users/nike/near/near-sdk-rs/near-sdk" }
```
Comment (willemneal) [inline runtime/near-vm-runner-standalone/example-contract/Cargo.toml:16]: Is this a local path?  Doesn't the params estimator clone this repo?

Comment (olonho) [reply runtime/near-vm-runner-standalone/example-contract/Cargo.toml:16]: Yeah, that fix shalln't be committed, it's a local testing artifact.

File: runtime/near-vm-logic/src/logic.rs [hunk 2]
```diff
@@ -2326,6 +2326,25 @@ impl<'a> VMLogic<'a> {
...
         self.gas_counter.pay_base(contract_compile_base)
     }
+
+    pub fn syscall(&mut self, syscall: u32,
+                   arg0: u64, arg1: u64, _arg2: u64, _arg3: u64,
+                   _arg4: u64, _arg5: u64, _arg6: u64, _arg7: u64) -> Result<u64> {
+        match syscall {
+            1 => {
+                println!("syscall #1: {}", arg0);
+                Ok(0)
```
Comment (ailisp) [inline runtime/near-vm-logic/src/logic.rs:2336]: Where is syscall actually happens? Here is only a print

**Review by evgenykuzyakov — CHANGES_REQUESTED** [review]
What's the point of this PR?

**Comment by bowenwang1996** [comment]
@olonho should we close this PR?

**Comment by matklad** [comment]
Discussed with olonho , closing for now!

**Pull request closed without merging**